src/QMCDrivers/Optimizers/DescentEngine.h

```diff
@@ -30,113 +32,204 @@ class DescentEngine
   typedef qmcplusplus::QMCTraits::FullPrecRealType FullPrecRealType;
 
 private:
-  //Vectors and scalars used in calculation of averaged derivatives in descent
+  // Vectors used in calculation of averaged parameter derivatives descent
+  // descent
   std::vector<FullPrecValueType> avg_le_der_samp_;
   std::vector<std::vector<FullPrecValueType>> replica_le_der_samp_;
 
   std::vector<FullPrecValueType> avg_der_rat_samp_;
   std::vector<std::vector<FullPrecValueType>> replica_der_rat_samp_;
 
-  FullPrecValueType w_sum_;
-  FullPrecValueType e_avg_;
-  FullPrecValueType e_sum_;
-  FullPrecValueType e_square_sum_;
-  FullPrecValueType e_square_avg_;
-
-  //Vector that stores the final averaged derivatives of the cost function
+  std::vector<FullPrecValueType> avg_numer_der_samp_;
+  std::vector<std::vector<FullPrecValueType>> replica_numer_der_samp_;
+
+  std::vector<FullPrecValueType> avg_denom_der_samp_;
+  std::vector<std::vector<FullPrecValueType>> replica_denom_der_samp_;
+
+  // Scalars used in calculation of average energy, variance, and target
+  // function for descent
+  ValueType w_sum_;
+
+  ValueType e_avg_;
+  ValueType e_var_;
+  ValueType e_sd_;
+  ValueType e_err_;
+
+  ValueType numer_avg_;
+  ValueType numer_var_;
+  ValueType numer_err_;
+
+  ValueType denom_avg_;
+  ValueType denom_var_;
+  ValueType denom_err_;
+
+  ValueType target_avg_;
+  ValueType target_var_;
+  ValueType target_err_;
+
+  // history of sampled |value/guiding|^2 ratios for one iteration
```

Comment:
`//` is for generic source code commenting and omitted by by doxygen. use `///` to document variables and functions. Please update throughout this file.